operators/src/concurrency/thread_pool.rs

```diff
@@ -144,22 +140,32 @@ impl ThreadPool {
     }
 }
 
+pub trait ThreadPoolContextCreator {
+    fn create_context(&self) -> ThreadPoolContext;
+}
+
+impl ThreadPoolContextCreator for Arc<ThreadPool> {
```

Comment:
no, the operator gets a `ThreadPoolContext` from the `ExecutionContext`. The context holds the `Arc` but it is private. The operator can only call compute on the context but not create a new context. Of course it could create a completely new `ThreadPool`, we have to trust that it doesn't do so.